torchao/prototype/mx_formats/constants.py

```diff
@@ -5,10 +5,12 @@
 # LICENSE file in the root directory of this source tree.
 import torch
 
+from torchao.utils import TORCH_VERSION_AT_LEAST_2_8
+
 # This is conceptually an enum of non-core dtypes
 # TODO(future PR): change to a cleaner way to represent this without
 # regressing torch.compile and while keeping things readable.
-DTYPE_FP4 = "fp4_e2m1"
+DTYPE_FP4 = torch.float4_e2m1fn_x2 if TORCH_VERSION_AT_LEAST_2_8 else "fp4_e2m1"
```

Comment:
1. we don't need to support old PyTorch versions for this directory, let's just move the code to the new version
2. we don't need `DTYPE_FP4` anymore and can replace its callsites by the actual dtype